esphome/components/tuya/number/__init__.py

```diff
@@ -33,6 +44,18 @@ def validate_min_max(config):
             cv.Required(CONF_MIN_VALUE): cv.float_,
             cv.Required(CONF_STEP): cv.positive_float,
             cv.Optional(CONF_MULTIPLY, default=1.0): cv.float_,
+            cv.Optional(CONF_DATAPOINT_HIDDEN): cv.All(
+                cv.Schema(
+                    {
+                        cv.Required(CONF_DATAPOINT_TYPE): cv.enum(
+                            DATAPOINT_TYPES, lower=True
+                        ),
+                        cv.Optional(CONF_INIT): cv.All(
+                            cv.Schema({cv.Required(CONF_VALUE): cv.float_})
+                        ),
```

Comment:
The extra nested config here seems over the top
```suggestion
                        cv.Optional(CONF_INITIAL_VALUE): cv.float_,
```

Reply:
This wasn't done by mistake. I plan to extend the yaml later with an option to restore the initial value from persistent storage, instead of a constant.

```yaml
init:
  - value: 10
  - restore_value: true # this part will be added in a separate PR
```

If I have the nested config now, the yaml will remain backwards compatible after I add the restoring part.

Reply:
In other places in ESPHome we have `initial_value` and `restore_value` as two separate keys beside each other.